core/src/main/scala/kafka/server/AlterPartitionManager.scala

```diff
@@ -263,7 +286,11 @@ class DefaultAlterPartitionManager(
         val partitionResponses = new mutable.HashMap[TopicPartition, Either[Errors, LeaderAndIsr]]()
         data.topics.forEach { topic =>
           topic.partitions.forEach { partition =>
-            val tp = new TopicPartition(topic.name, partition.partitionIndex)
+            val tp = new TopicPartition(
+              topicNamesByIds.getOrElse(topic.topicId, topic.topicName),
```

Comment:
I guess we could get this from the MetadataCache as well. Also, I wonder if we should validate `topicName` is non-empty before using it?